inference-engine/src/inference_engine/ie_compound_blob.cpp

```diff
@@ -192,6 +192,50 @@ void verifyI420BlobInput(const Blob::Ptr& y, const Blob::Ptr& u, const Blob::Ptr
     }
 }
 
+TensorDesc verifyBatchedBlobInput(const std::vector<Blob::Ptr>& blobs) {
+    // verify invariants
+    if (blobs.size() < 1) {
+        THROW_IE_EXCEPTION << "at least 1 sub-blob needed";
+    }
+
+    const auto subBlobDesc = blobs[0]->getTensorDesc();
+    if (std::any_of(blobs.begin(), blobs.end(),
+                    [&subBlobDesc](const Blob::Ptr& blob) {
+                        return blob->getTensorDesc() != subBlobDesc;
```

Comment:
`TensorDesc` also includes `BlockingDesc`, which contains such properties as strides and offsets. IMO, we should allow such properties to be different for blobs in batch. For example, batch of ROI blobs, where offset can be different. @ilyachur @ilya-lavrenov What do you think?

Reply:
This check might be less restrictive, but it leads the question: how to build tensor descriptor of the `BatchedBlob` instance? Which sub-blob to be selected as the template resulting `TensorDesc`?

Anyway, there is defined another one .ctor which allows to bypass this verification at all.

Reply:
From my point of view we have 2 possible solutions in this case:

- `TensorDesc` in `BatchedBlob` and sub-blobs should be the same.
- If sub-blobs can be ROI blobs (with custom strides and offsets), I think that `BatchedBlob` should contain the planar `TensorDesc` without any custom strides or offsets in the `BlockingDesc`. But users should know about this behaviour and process each sub-blob separatelly.

Reply:
> If sub-blobs can be ROI blobs (with custom strides and offsets), I think that `BatchedBlob` should contain the planar `TensorDesc` without any custom strides or offsets in the `BlockingDesc`.

Which `Layout` value should be set for resulting `TensorDesc` in such case?

Reply:
@ilyachur , can you please response? Thank you.

Reply:
I propose to use the default (for OpenVINO layout) NCHW.

Reply:
Should we expect equivalent shapes (dimensions) and data types for all ROI blobs?\
If not, there is the new open: which shape and data type to be generated?

P.S. Why we cannot delegate `TensorDesc`  generation to the client code?
 - see: https://github.com/openvinotoolkit/openvino/pull/2203/files#diff-0dd753c1fc899092d88a6e0a10c5664523c26ad433b67106a9a329f6e9802e47R323

Reply:
In this case if shapes and data type can be different we cannot use common `TensorDesc`. 
Should we use an empty tensor desc in this case?

Or if we have some steps which allow to align shapes and data types to one value, we can use this value for common TensoeDesc.

Reply:
For NV12 blob we don't have valid getTensorDesc, it's empty. I think we can do the same for BatchedBlob if blobs in batch have different parameters (and it's valid use case for VPUX, where different blobs in batch are just ROIs from bigger picture)

Reply:
This PR also contains modification for NV12 and I420 to set proper `TensorDesc` which is logically describe the shape and data type of images.

Reply:
Offline review outcome: keep this verification as is.

New change to be prepared for  other cases (aka ROI)